packages/ra-ui-materialui/src/layout/NotFound.tsx

```diff
@@ -45,8 +46,14 @@ function goBack() {
     window.history.go(-1);
 }
 
-const NotFound = props => {
-    const { className, classes: classesOverride, title, ...rest } = props;
+interface Props extends RouteComponentProps {
+    className: string;
+    title: TitleProps['defaultTitle'];
+    classes?: object;
```

Comment:
I think MUI offers a way to infer the accepted type based on the makeStyles.

Reply:
I don't know about it but something like ReturnType<typeof useStyles> should work?

Reply:
absolutely!

Reply:
No, wait, it won't work. the `classes` *argument* passed to `useStyles` isn't the same type as the *result* of `useStyles`, which is usually named `classes`.

I'm looking for the right way to type the `classes` override (the material-ui documentation doesn't mention it).

Reply:
Okay, there is no easy way to type these - see my comment in https://github.com/mui-org/material-ui/issues/17973#issuecomment-638868260.

Let's default to `any` for now.

Reply:
Or we can add a generic type to `ra-ui-material-ui`:

```tsx
import { CSSProperties, CreateCSSProperties } from "@material-ui/styles";

type PropsFunc<Props extends object, T> = (props: Props) => T;

export type ClassesOverride<
  UseStyles extends (props: any) => Record<string, string>,
  Props extends object = {}
> = Partial<
  {
    [p in keyof ReturnType<UseStyles>]:
      | CSSProperties
      // JSS property bag where values are based on props
      | CreateCSSProperties<Props>
      // JSS property bag based on props
      | PropsFunc<Props, CreateCSSProperties<Props>>
  }
>;
```

which allows to type our `classes` props as follows:

```tsx
interface Props {
    classes: ClassesOverride<typeof useStyles>;
    ...
}
```

What do you think?

Reply:
Excellent but from what I understand the parent component passes a `classes` object each key is a `className` string, not `CSSProperties` object?

```jsx
import { makeStyles } from '@material-ui/core';
import React, { FC } from 'react';

const useNotFoundStyles = makeStyles(
	theme => ({
		root: {
			display: 'flex',
		},
	}),
	{ name: 'RaNotFound' },
);

type ClassesOverride<
	UseStyles extends (props: any) => Record<string, string>
> = Partial<Record<keyof ReturnType<UseStyles>, string>>;

interface NotFoundProps {
	classes?: ClassesOverride<typeof useNotFoundStyles>;
}

export const NotFound: FC<NotFoundProps> = props => {
	const classes = useNotFoundStyles(props);
	return <div className={classes.root}>DEMO</div>;
};
```

```jsx
const useLayoutStyles = makeStyles(
  theme => ({
    root: {
      display: "block"
    }
  }),
  { name: "RaLayout" }
);

interface LayoutProps {}

export const Layout: React.FC<LayoutProps> = props => {
  const classes = useLayoutStyles(props);
  return (
    <div>
      <NotFound classes={{ root: classes.root }} />
    </div>
  );
};
```

Reply:
Totally, you're right and I was too tired when writing my comment. Sorry for that!

Reply:
No worries you are working on so many things, I'll add this type to Layout components for now and start a new PR for all `ra-material-ui` other components.